### DISCUSSION TITLE
Understand this entire codebase with the help of a custom GPT [LLM-C by Andrej Karpathy]

### DISCUSSION THREAD

**Discussion by ehzawad** [karpathy/llm.c#158] [General]
GPT Link: [LLM-C by Andrej Karpathy](https://chat.openai.com/g/g-ZFo5NRSJS-llm-c-by-andrej-karpathy)

I felt like it could helpful for someone who wanted to understand this codebase (every function, including mathematical expression) in details, without tediously uploading every source file!

This GPT will explain everything from this repo, including Makefile, requirements.txt, just ask away!

The repo is constantly updating, but this code is updated to the last commit of the main branch `9e71af96784f11d7b234e263ed7e8ea933b1cc4a`

**Comment by ehzawad** [comment]
It has now been updated to commit **d2482ff1ad055b394279f61b418a4d2131a080f2** (HEAD -> master, origin/master, origin/HEAD)`

Since GPT-4o has come, the custom GPT behavior has changed a little imo!

**Comment by karpathy** [comment]
It's probably better than nothing but it's not very good. E.g. it mixes up that dev/cuda is just dev scratch space not "prod code", so when you ask it about gelu it pulls up an example from dev/cuda, which is not super ideal. but ok

**Reply by ehzawad** [reply]
Thanks for the feedback. Honestly, I didn't extensively test it out myself to see how it performs! But I think it doesn't mess up with "gelu" anymore!

**Reply by wbmlr** [reply]
Are there alternatives that people without a premium ChatGPT subscription can use?
